java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java

```diff
@@ -1468,4 +1467,10 @@ public void getClass_return_type() {
     assertThat(((WildCardType) param).bound.is("ISuiteListener")).isTrue();
   }
 
+  @Test
+  public void lambda_on_sam_interface_defining_object_methods() throws Exception {
+    Result result = Result.createFor("SAMResolution");
+    assertThat(((VariableTree) result.symbol("var").declaration()).initializer().symbolType().is("MyVar"));
+    assertThat(((VariableTree) result.symbol("var2").declaration()).initializer().symbolType().is("MyVar2"));
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Complete the assertion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2970)